packages/api-v2/src/APIProvider.tsx

```diff
@@ -68,148 +42,145 @@ const getWebSocketURL = () => {
+            }
+        );
+    }
 
-    if (!standaloneDerivAPI.current)
-        standaloneDerivAPI.current = standalone ? initializeDerivAPI(() => setReconnect(true)) : null;
+    useEffect(() => {
+        return () => {
+            derivAPIRef.current.disconnect();
+            reactQueryRef.current?.clear();
+        };
+    }, []);
 
     const setOnReconnected = useCallback((onReconnected: () => void) => {
         onReconnectedRef.current = onReconnected;
     }, []);
 
+    const setOnConnected = useCallback((onConnected: () => void) => {
+        if (isOpenRef.current) {
+            onConnected();
+        } else {
+            onConnectedRef.current = onConnected;
+        }
+    }, []);
+
     const send: TSendFunction = (name, payload) => {
-        return standaloneDerivAPI.current?.send({ [name]: 1, ...payload });
+        return derivAPIRef.current?.send({ [name]: 1, ...payload });
     };
 
     const subscribe: TSubscribeFunction = async (name, payload) => {
         const id = await hashObject({ name, payload });
-        const matchingSubscription = subscriptions.current?.[id];
+        const matchingSubscription = subscriptionsRef.current?.[id];
         if (matchingSubscription) return { id, subscription: matchingSubscription };
 
         const { payload: _payload } = payload ?? {};
 
-        const subscription = standaloneDerivAPI.current?.subscribe({
+        const subscription = derivAPIRef.current?.subscribe({
             [name]: 1,
             subscribe: 1,
             ...(_payload ?? {}),
         });
 
-        subscriptions.current = { ...(subscriptions.current ?? {}), ...{ [id]: subscription } };
+        subscriptionsRef.current = { ...(subscriptionsRef.current ?? {}), ...{ [id]: subscription } };
         return { id, subscription };
     };
 
     const unsubscribe: TUnsubscribeFunction = id => {
-        const matchingSubscription = subscriptions.current?.[id];
+        const matchingSubscription = subscriptionsRef.current?.[id];
         if (matchingSubscription) matchingSubscription.unsubscribe();
     };
 
     useEffect(() => {
-        const currentDerivApi = standaloneDerivAPI.current;
-        const currentSubscriptions = subscriptions.current;
+        const currentderivAPIRef = derivAPIRef.current;
+        const currentsubscriptionsRef = subscriptionsRef.current;
 
         return () => {
-            if (currentSubscriptions) {
-                Object.keys(currentSubscriptions).forEach(key => {
-                    currentSubscriptions[key].unsubscribe();
+            if (currentsubscriptionsRef) {
+                Object.keys(currentsubscriptionsRef).forEach(key => {
+                    currentsubscriptionsRef[key].unsubscribe();
                 });
             }
-            if (currentDerivApi && currentDerivApi.connection.readyState === 1) currentDerivApi.disconnect();
+            if (currentderivAPIRef && currentderivAPIRef.connection.readyState === 1) currentderivAPIRef.disconnect();
```

Comment:
Nitpicking: casing in `currentderivAPIRef` and `currentsubscriptionsRef`

Reply:
thanks, I've added it to my todo list in next PR which I'm already working on,